game.py

```diff
@@ -0,0 +1,148 @@
+                input()
+            print(f"\t\t\t\t\t\t\t\tTeam {self.team}.")
+            print(f"\t\t\t\t\t\t\tThe code word is --{self.code_word.capitalize()}--\n\t\t\t\t\t\t\t\tfor --{self.code_number}--\n")
+
+            flag = False
+            # while loop is used to check if the guess exists on the board and if it has already been guessed or not.
+            while not flag:
+                self.guess = input("Enter the word on the card you think this applies to: ").capitalize()
+                if self.guess in [card.word for card in self.board.cards if card.flipped]:
+                    print("This has already been guessed. Guess again.\n")
+                    continue
+                for card in self.board.cards:
+                    if self.guess == card.word:
+                        flag = True
+                        break
+                if not flag:
+                    print("Please enter a word on one of the cards. Guess again\n")
+                if flag:
+                    break
+            system("clear")
+        else:
+            input()
+    
+    def check_guess(self):
+        '''There are 4 options.
+        1)It is their teams word. 2)It is the other team's word 
+        3)It is a nuetral word 4)It is the death card and the game is over.'''
+        for card in self.board.cards:
+            if self.guess == card.word:
+                card.flipped = True
+                if self.team == card.color:
+                    self.guess_value = 1
+                elif self.team != card.color and card.color != "yellow" and card.color != "black":
+                    self.guess_value = 2
+                elif card.color == "yellow":
+                    self.guess_value = 3
+                elif card.color == "black":
+                    self.guess_value = 4
+                    self.game_over = True
+                break
+        if self.guess_value == 1:
+            self.code_number -= 1
+        else:
+            self.code_number = 0
+
+    def check_win(self):
+        '''Function ends the game on a few conditions:
+        1) If Blue or Red pick the death card the game will end.
+        2) If either team gets their entire word set the game will end.'''
+        color = ["Blue","Red"]
+
+        if self.game_over and not self.skip:
+            print(f"{self.team} picked the death card.")
+            print("The game is over!")
+            print(f"{color[0] if self.team.capitalize() == color[1] else color[1]} wins! Congratulations!")
+
+        # This runs twice in order to check the red and the blue decks
+        for i in color:
+            all_team_color_cards = [card.word for card in self.board.cards if card.color.capitalize() == i]
+            all_flipped_team_color_cards = [card.word for card in self.board.cards if card.flipped and card.color.capitalize() == i]
+
+            if all_team_color_cards == all_flipped_team_color_cards:
+                print("The game is over!")
```

Comment:
You will want to move all the print statements in `check_win` to the `render` function. You should be able to use things like `self.game_over`, `self.team`, and `self.guess_value` in the `render()` function to print the appropriate winner and message.

Your state update type functions should not have any calls to `input()` or `print()`. They should just be about changing the variables that make up the state of your game. This will allow you to fully test them with automated testing (given a certain input state, do you get the expected output state after you call X function?).

It will also make the initial Python to C port easier, as we will start by just porting the state update type functions to C. That way you don't have to worry about keeping track of which instructions are printed in Python code vs C code.